### PR TITLE
Object compliance unit tests

### LINKED ISSUE #54: Test_Toolkit: Write the unit tests
Write the unit tests that perform each individual check for each test.

### PR DESCRIPTION
**Pull request by FraserGreenroyd**
### Issues addressed by this PR
Fixes #54 

 ### Test files

 ### Changelog

 ### Additional comments

### CODE REVIEW

File: Test_oM/Error.cs [hunk 1]
```diff
@@ -34,5 +34,7 @@ public class Error : BHoMObject
         public ErrorLevel Level { get; set; }
         public string Message { get; set; }
         public Location Location { get; set; }
+
+        private double DeliberateFail { get; set; }
```
Comment (awakeman) [inline Test_oM/Error.cs]: This is appropriately failing the test. Whilst it is demonstrating the test works it needs fixing now.

Comment (FraserGreenroyd) [reply Test_oM/Error.cs]: Howzat :wink:

**Comment by FraserGreenroyd** [comment]
Closing for the new PR which is cleaner but keeping this available to review tests later.

**Pull request closed without merging**